src/validation/stmt_validator.rs

```diff
@@ -337,14 +335,29 @@ impl StatementValidator {
 
     fn validate_reference(
         &mut self,
-        id: &usize,
+        statement: &AstStatement,
         ref_name: &str,
         location: &SourceRange,
         context: &ValidationContext,
     ) {
-        if !context.ast_annotation.has_type_annotation(id) {
+        // unresolved reference
+        if !context.ast_annotation.has_type_annotation(statement) {
             self.diagnostics
                 .push(Diagnostic::unresolved_reference(ref_name, location.clone()));
+        } else if let Some(StatementAnnotation::Variable {
+            qualified_name,
+            variable_type,
+            ..
+        }) = context.ast_annotation.get(statement)
+        {
+            if variable_type.is_private()
+                && !qualified_name.starts_with(context.qualifier.unwrap_or(".INVALID."))
```

Comment:
What's the invalid thing?

Reply:
nvm, I see now you are creating it for the unwrap.
I wonder if there's a better way of doing that? but it's fine for now

Reply:
yes it's really strange ... i'd like something like
myOptional.test(|it| qualified_name.starts_with(it)) which is false, if the optional is empty. Maybe i should do a helper-method?